modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/IgniteIndexScan.java

```diff
@@ -39,13 +41,18 @@
     /** */
     private final long sourceId;
 
+    /** */
+    private final RelCollation idxCollation;
+
     /**
      * Constructor used for deserialization.
      *
      * @param input Serialized representation.
      */
     public IgniteIndexScan(RelInput input) {
-        super(changeTraits(input, IgniteConvention.INSTANCE));
+        super(changeTraits(input, IgniteConvention.INSTANCE, input.getCollation()));
```

Comment:
I don't sure it's a good idea to restore traits after deserialization. We either should do this for every node or don't do it at all.
Well, there is a few nodes (e.g. Exchange) which have an assertion inside their constructor, so `changeTraits` is used to overcome this problem. But for other cases I would prefer to avoid using it

Reply:
Here we need information about collation on remote nodes, for other rels we don't need such information.

Reply:
Sort relations don't have projections and "required columns", so sorting collation and trait collation are the same here. For index scans index collation and trait collation is different and there is confusion. Perhaps, it's better to store index collation in the "collation" field and restore output collation on remote nodes by index collation and projects. It's much easier than restoring index collation by bounds as currently implemented in this patch. WDYT?

Reply:
> Perhaps, it's better to store index collation in the "collation" field and restore output collation on remote nodes by index collation and projects. 

Hmm, but in this case sort node will be inserted even if it's not required (for example, when we just filter values by index), looks like it's not an optimal solution.